lib/construct.ts

```diff
@@ -435,8 +436,8 @@ export class Node {
 
     // since constructs can be added to the tree during invokeAspects, call findAll() to recreate the list.
     // use PREORDER.reverse() for backward compatability
-    for (const construct of this.findAll(ConstructOrder.PREORDER).reverse()) {
-      if (construct instanceof Construct) {
+    for (const construct of this.findAll(ConstructOrder.PREORDER).reverse()) {      
+      if (Construct.isConstruct(construct)) {
```

Comment:
I am not sure this change makes sense. Technically `this.findAll` returns an array of `Construct` objects.

Since this is dynamic anyway, I wonder if actually the correct fix here should be:

```ts
const cn = construct as any;
if ('onPrepare' in cn) {
  if (typeof(cn.onPrepare) !== 'function') { throw new Error(`expecting "onPrepare" to be a function`); }
  cn.onPrepare();
}
```

Reply:
I tried to be conservative after checking the old code in cdk, lgtm i will change it

Reply:
I didn't have enough time to think about writing a test that mutates the prototipe of a construct just to raise the issue, I will appreciate some help.
Of course I will write the tests to ensure that all the onPrepare functions will be run, and the others use cases

Reply:
test added